ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -1732,9 +1729,18 @@ module VM = struct
             Domain.destroy task ~xc ~xs ~qemu_domid ~vtpm:(vtpm_of ~vm)
               ~dm:(dm_of ~vm) domid ;
             (* Detach any remaining disks *)
+            let dps =
+              List.map
+                (fun device ->
+                  ( Device.Generic.get_private_key ~xs device _dp_id
+                  , Storage.vm_of_domid (Some domid)
```

Comment:
This is still a constant value for all dps. There is no need to put it inside the list. Just leave dps as it was and pass `Storage.vm_of_domid (Some domid)` to `Storage.dp_destroy` below.